### PR TITLE
Refactor Message handling

### PR DESCRIPTION
**Pull request by Apollon77**
I refactored the message handling in a way that the handling is more near where also the logic is and reduce duplicate code.

Convetion is "controller*" will be handled by Controller class and "device*" by the device class

### CODE REVIEW

**Review by foxriver76 — APPROVED** [review]
lgtm just small suggestions

File: src/main.ts [hunk 1]
```diff
@@ -150,100 +144,78 @@ export class MatterAdapter extends utils.Adapter {
...
+                try {
+                    const result = await this.controller.handleCommand(obj.command, obj.message);
+                    if (result !== undefined) {
+                        obj.callback && this.sendTo(obj.from, obj.command, result, obj.callback);
+                    }
+                } catch (error) {
+                    this.log.warn(`Error while handling command ${obj.command} for controller: ${error.stack}`);
+                    obj.callback && this.sendTo(obj.from, obj.command, { error: error.message }, obj.callback);
                 }
+            } else {
```
[the remark this answers is not in the document]
Comment (Apollon77) [reply src/main.ts:165]: agreed, I kill all places

File: src/main.ts [hunk 2]
```diff
@@ -767,15 +741,15 @@ export class MatterAdapter extends utils.Adapter {
...
+            if (!devices.find(obj => obj._id === deviceId)) {
                 await device.stop();
                 this.devices.delete(deviceId);
             }
         }
 
         // Create new devices
-        for (const d in _devices) {
-            const device = _devices[d];
+        for (const d in devices) {
```
[the remark this answers is not in the document]
Comment (Apollon77) [reply src/main.ts:751]: right, missed that one

File: src/matter/ControllerNode.ts [hunk 3]
```diff
@@ -56,6 +57,12 @@ interface AddDeviceResult {
     nodeId?: string;
 }
 
+interface EndUserCommissioningOptions {
+    qrCode: string | undefined;
```
[the remark this answers is not in the document]
Comment (Apollon77) [reply src/matter/ControllerNode.ts:61]: agreed

File: src/matter/ControllerNode.ts [hunk 4]
```diff
@@ -110,33 +116,62 @@ class Controller {
...
-         *
-         * Note: This example also uses the initialized storage system to store the device parameter data for convenience
-         * and easy reuse. When you also do that, be careful to not overlap with Matter-Server own contexts
-         * (so maybe better not ;-)).
-         */
+        if (this.parameters.ble) {
+            try {
+                Ble.get = singleton(() => new BleNode({ hciId: this.parameters.hciId }));
+            } catch (error) {
+                this.adapter.log.info(`Failed to initialize BLE: ${error.message}`);
```
[the remark this answers is not in the document]
Comment (Apollon77) [reply src/matter/ControllerNode.ts:123]: ok i do warn

**Pull request merged**